src/TenantCloud/APIVersioning/Constrain/SemanticConstraintChecker.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace TenantCloud\APIVersioning\Constrain;
+
+use TenantCloud\APIVersioning\Version\LatestVersion;
+use TenantCloud\APIVersioning\Version\SemanticVersion;
+use TenantCloud\APIVersioning\Version\Version;
+use Tests\SemanticConstraintCheckerTest;
+
+/**
+ * @see SemanticConstraintCheckerTest
+ */
+class SemanticConstraintChecker implements ConstraintChecker
+{
+	public function __construct(private readonly StringConstraintParser $constraintParser)
+	{
+	}
+
+	/**
+	 * If we have version rule like '<=3.0' and '<=2.0' and got version from client '1.0'
+	 * we should suggest only one controller function for this request.
+	 * So we suggest which controller should run.
+	 * If we got exactly version (for ex. rule is ==2.0) and we got version from client 2.0
+	 * we suggest this controller as well
+	 *
+	 * Examples :
+	 * matches('3.0', ['==1.0', '<=2.0', '<=3.0'])) => '<=3.0' (got max available version rule)
+	 * matches('2.0', ['==1.0', '==2.0', '<=3.0'])) => '==2.0' (got exactly version rule)
+	 * matches('3.0', ['==1.0', '==2.0', '<3.0'])) => null (no rule find. Got empty string)
+	 */
+	public function matches(Version $version, array $constraints): ?Constraint
+	{
+		$result = [];
+
+		// If latest version - create the highest semantic version object.
+		if ($version instanceof LatestVersion) {
+			$version = new SemanticVersion('1000000.0');
```

Comment:
Maybe 1000000.0 as a constant?

Reply:
will add latestVersionResolver for this cases. We can have multiple realisation of version (date, semantic, etc...)